gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/UploadService.java

```diff
@@ -45,26 +48,31 @@ public DocumentInstance uploadFile(MultipartFile file, UUID organizationId, UUID
         DocumentInstance instance =
                 documentService.getDocumentByFileName(projectId, organizationId, fileName);
 
+        // find agent's userID
+        UUID userId = projectAgentService.getAgentByProjectId(projectId).getUserId();
```

Comment:
the userID can be extracted directly from the project by by getting the projectAgent and then the userID. Therefore there is no need to autowire the projectAgentService